Tasks/BashV3/bash.ts

```diff
@@ -72,7 +73,20 @@ async function run() {
                 targetFilePath = input_filePath;
             }
 
-            contents = `. '${targetFilePath.replace("'", "'\\''")}' ${input_arguments}`.trim();
+            if (old_source_behavior) {
+                contents = `. '${targetFilePath.replace("'", "'\\''")}' ${input_arguments}`.trim();
+            } else {
+                // Check if executable bit is set
+                let stats: fs.Stats = tl.stats(input_filePath);
+                if ((stats.mode & 1) > 0) {
```

Comment:
Can we add a plain English description of what this is doing :)

Reply:
Sure, done